webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -7313,7 +7355,44 @@ var CURRENT_GROUP_DATA={};
                 };
                 return conversationDetail;
             };
-
+            _this.getBotDetails = function () {
+                window.Applozic.ALApiService.ajax({
+                    url: MCK_BOT_API + "/bot/" + MCK_APP_ID,
+                    type: 'get',
+                    global: false,
+                    success: function (data) {
+                    var flag = false;
+                    console.log(data);
+                    for (var key in data)
+                    {
+                        if (data.hasOwnProperty(key)) {
+                            if(data[key].name == CURRENT_GROUP_DATA.conversationAssignee){
+                                //console.log(data[key].name);
+                                if(data[key].aiPlatform == 'dialogflow')
+                                {
+                                    if(!data[key].autoHumanHandoff){
+                                        CHAR_CHECK = true;
```

Comment:
@hardikparashar there can be multiple conversations with different bots so CHAR_CHECK can't be a global variable, its scope needs to be limited for one group only.

Reply:
CHAR_CHECK gets changed for every new conversation window opened

Reply:
@hardikparashar still it can't be a global variable, if there are 4 conversations, there are 4 possible values.

Reply:
Are the conversations opened simultaneously in the same window?

Reply:
no, but this is API call which is Async. Lets say a user opens one conversation then API CALL-1 starts, users presses back and then opens another conversation, API CALL-2 starts now. There is no guarantee that API CALL-1 will finish before API CALL-2 too, what if API CALL-2 finishes first and then API CALL-2 completes and override the value for CHAR_CHECK

Reply:
Okay I understand now

Reply:
resolved in new commit